sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -16,13 +16,14 @@ internal class DiagnosticScopeFactory
         private static Dictionary<string, DiagnosticListener>? _listeners;
         private readonly string? _resourceProviderNamespace;
         private readonly DiagnosticListener? _source;
-        private readonly bool _suppressNestedClientScopes;
+        private readonly bool _suppressNestedClientActivities;
 
-        public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNamespace, bool isActivityEnabled, bool suppressNestedClientScopes = true)
+        public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNamespace, bool isActivityEnabled, bool suppressNestedClientActivities)
```

Comment:
Just curious why we are removing the default.

Reply:
similar to above: 

If we keep default here, it should be consistent with `ClientDiagnostics` and would be suppression *off*.

`ClientDiagnostics` needs suppression *off* because all previous autogenerated code needs to be backward compatible. 

But the right value for all future manually written clients would be suppression *on*.

If we're ok with having inconsistent default values, I'm happy to return default `= true` back, otherwise, I prefer to force callers to make a choice